parser/src/parser.rs

```diff
@@ -71,13 +155,36 @@ impl<'a> Parser<'a> {
         Ok(())
     }
 
-    /// Parses the tokens into an abstract syntax tree.
+    //simple error report system, should be enhanced in later PRs
+    pub(crate) fn report_error(&self, err: &ParseError) {
+        eprintln!("ERROR !");
+        eprintln!("message: {}", err.message)
+    }
+
+    //traverse current expression and go to next expression
+    fn repos_to_next_expr(&mut self) {
+        while self.cursor.lookahead(eox()).is_none() {
+            self.cursor.advance(next());
+        }
+    }
+
+    /// Parses input tokens into an abstract syntax tree representation.
     pub(crate) fn parse(&mut self) -> ParseResult<Vec<Expr<'a>>> {
         let mut statements = Vec::new();
 
         while !self.cursor.is_at_end() {
-            statements.push(self.statement()?);
-            self.cursor.advance(space().then(eox()));
+            let statement = self.statement();
+
+            if let Err(error) = &statement {
+                self.report_error(error);
+                self.repos_to_next_expr();
+            }
```

Comment:
Errors should not be printed here, and parsing errors are most likely unrecoverable.

Reply:
I've been thinking about this again, certain parse errors are totally recoverable, so having a repos method is totally fine. The first part of the sentence is still valid.